packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -232,6 +233,12 @@ function extendLocalReporterToCatchPluginErrors({
     error,
     panic,
     panicOnBuild,
+    deprecate: args => {
+      reporter.deprecate({
+        ...args,
+        pluginName: isDefaultPlugin ? pluginName : pluginName,
```

Comment:
Probably should be
```suggestion
        pluginName: isDefaultPlugin ? `Your site's "gatsby-node.js"` : pluginName,
```

Reply:
Actually probably not because setting this up will result in weird message together with:
https://github.com/gatsbyjs/gatsby/blob/1f5be8a148196abdbccff4eb8767daf87b2c1057/packages/gatsby-cli/src/reporter/loggers/ink/components/deprecation.tsx#L15-L17

so I will likely set just `pluginName` and instead change how it's rendered in `Deprecation` ink component so it doesn't say
> Plugin: Your site's "gatsby-node.js" - upgrade to the latest version or contact the plugin author.

because that doesn't make much sense and having metadata with pluginName set to `default-site-plugin` is good enough